src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -0,0 +1,120 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.PreparedQuery.TooManyResultsException;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for creating new recipes. */
+@WebServlet("/edit-recipe")
+public class EditRecipeServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException  {
+    // get edits from form
+    String idResponse = request.getParameter("recipeID");
+    String name = request.getParameter("edit-name");
+    String tagsResponse = request.getParameter("edit-tags");
+    String description = request.getParameter("edit-description");
+    String ingredientsResponse = request.getParameter("edit-ingredients");
+    String stepsResponse = request.getParameter("edit-steps");
+    // privacy will be used once we give users the option to make their recipes public
+    // String privacy = request.getParameter("privacy");
+
+    // declaring the lists
+    ArrayList<String> tags;
+    ArrayList<String> ingredients;
+    ArrayList<String> steps;
+
+    long id = Long.parseLong(idResponse);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Key key = KeyFactory.createKey("PrivateRecipe", id);
+    try {
+      Entity recipeEntity = datastore.get(key);
+
+      // if the name input in the form is empty, the original value will be saved
+      if (name.equals("")) {
```

Comment:
What if you initialized all the variables to be the data from Datastore, and then only have the if-statements to change the values if an empty string was not found? This could help shorten the code and remove some "else" blocks.